### PR TITLE
{math} [goolf/1.5.16, intel/2018a] PaStiX 5.2.2.22, 6.0.0 #6115 (REVIEW)

### PR DESCRIPTION
**Pull request by SimonPinches**

### CODE REVIEW

File: easybuild/easyconfigs/p/PaStiX/PaStiX-5.2.2.22-goolf-1.5.16.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+toolchainopts = {'pic': True}
+
+source_urls = ['https://gforge.inria.fr/frs/download.php/file/35070/']
+sources = ['%(namelower)s_%(version)s.tar.bz2']
+checksums = [
+    '30f771a666719e6b116f549a6e4da451beabab99c2ecabc0745247c3654acbed',  # pastix_5.2.2.22.tar.bz2
+    '382311f53e855f19962abd0dfd4b543d734383f40f2fbeed586a2a6e1a6ff5d2',  # pastix-memory.patch
+]
+
+patches = ['pastix-memory.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/p/PaStiX/PaStiX-5.2.2.22-goolf-1.5.16.eb]: @SimonPinches Please move this up above `checksums`, and rename the patch file to something like `PaStiX-5.2.2.22_memory.patch`

File: easybuild/easyconfigs/p/PaStiX/PaStiX-5.2.2.22-goolf-1.5.16.eb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+dependencies = [
+    ('hwloc', '1.8.1', '', ('GCC', '4.8.3')),
```
Comment (boegel) [inline easybuild/easyconfigs/p/PaStiX/PaStiX-5.2.2.22-goolf-1.5.16.eb]: No need to hardcode to a specific subtoolchain (anymore)?

```python
('hwloc', '1.8.1'),
```

File: easybuild/easyconfigs/p/PaStiX/PaStiX-5.2.2.22-goolf-1.5.16.eb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+]
+
+skipsteps = ['configure']
+
+start_dir = 'src'
+
+prebuildopts = "cp -a config/LINUX-GNU.in config.in && "
+buildopts = 'BLASLIB="-L$BLAS_LIB_DIR $LIBBLAS" HWLOC_HOME=$EBROOTHWLOC SCOTCH_HOME=$EBROOTSCOTCH '
+# take control over compiler flags
+buildopts += 'CCFOPT="$CFLAGS -DMEMORY_USAGE -DWITH_SCOTCH" '
```
Comment (boegel) [inline easybuild/easyconfigs/p/PaStiX/PaStiX-5.2.2.22-goolf-1.5.16.eb]: Doesn't this make the patch file obsolete?

File: easybuild/easyconfigs/p/PaStiX/PaStiX-6.0.0-intel-2018a.eb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+toolchainopts = {'pic': True}
+
+source_urls = ['https://gforge.inria.fr/frs/download.php/file/37244/']
+sources = [SOURCELOWER_TAR_GZ]
+checksums = [
+    '1dc37475964d7baa0df30c5909bfdf459dd0db66b866f9d9f566ab6fd7e12d7f',  # pastix-6.0.0.tar.gz
+    '2b898a2bc8c5e97272fa61c8365f6332222813fc02ea4188cbad089e72f50e92',  # pastix-intel.patch
+]
+
+patches = ['pastix-intel.patch']
```
Comment (boegel) [inline easybuild/easyconfigs/p/PaStiX/PaStiX-6.0.0-intel-2018a.eb]: Please move up above `checksums`, and rename the patch file to something like `PaStiX-6.0.0_intel.patch`

File: easybuild/easyconfigs/p/PaStiX/PaStiX-6.0.0-intel-2018a.eb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+builddependencies = [
+    ('CMake', '3.9.1')
+]
+
+dependencies = [
+    ('hwloc', '1.11.8', '', ('GCCcore', '6.4.0')),
+    ('SCOTCH', '6.0.4', '', ('intel', '2018a')),
```
Comment (boegel) [inline easybuild/easyconfigs/p/PaStiX/PaStiX-6.0.0-intel-2018a.eb]: No need to hardcode toolchain for dependencies:

```python
dependencies = [
    ('hwloc', '1.11.8'),
    ('SCOTCH', '6.0.4'),
]

File: easybuild/easyconfigs/p/PaStiX/pastix-intel.patch [hunk 6]
```diff
@@ -0,0 +1,12 @@
+diff -Nru pastix-6.0.0-orig/cmake_modules/morse_cmake/modules/find/FindBLAS.cmake pastix-6.0.0/cmake_modules/morse_cmake/modules/find/FindBLAS.cmake
```
Comment (boegel) [inline easybuild/easyconfigs/p/PaStiX/pastix-intel.patch]: please include patch description + mention author (any lines above this line are comments)

and then don't forget to update the checksum for the patch file

File: easybuild/easyconfigs/p/PaStiX/pastix-memory.patch [hunk 7]
```diff
@@ -0,0 +1,12 @@
+diff -Nru pastix_5.2.2.22-orig/src/config/LINUX-GNU.in pastix_5.2.2.22/src/config/LINUX-GNU.in
```
Comment (boegel) [inline easybuild/easyconfigs/p/PaStiX/pastix-memory.patch]: please include patch description + mention author above (and update checksum accordingly)

File: easybuild/easyconfigs/p/PaStiX/PaStiX-5.2.2.22-goolf-1.5.16.eb (hunk 3 above)
Comment (SimonPinches) [reply easybuild/easyconfigs/p/PaStiX/PaStiX-5.2.2.22-goolf-1.5.16.eb]: No, these options are needed as well.

**Comment by migueldiascosta** [comment]
@SimonPinches, this PR is being closed for the following reason(s): uses a deprecated toolchain.
For more information on deprecated toolchains, see https://easybuild.readthedocs.io/en/latest/Deprecated-easyconfigs.html#deprecated-toolchains.
Please consider reopening or submitting a new PR using a supported toolchain.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**